### DISCUSSION TITLE
Not convinced this is ready for prime time

### DISCUSSION THREAD

**Discussion by genox** [swellstores/community#241] [Help] [answered]
Hi,

By chance I read in another post here that API requests are throttled on standard plan, checked back on the pricing page. The wording used on the page is "Stores on the Community plan will have API calls throttled with excessive usage. The Standard plan has higher allowances, and Enterprise plans have no API limits."

In the discussion here https://github.com/orgs/swellstores/discussions/238 it is impossible to figure out what the actual limits are, how they are handled, etc. This is very intransparent. It doesn't sound as the OP in this discussion is running a huge operation.

From my experience by just using it during development, it feels like the API is being throttled on standard plan by default. I previously thought this might be due to the fact that there are no EU datacenters in operation but now I have to consider that the limit is basically 0? Is that true? There is no way to see what's going on. So what defined excessive usage? And how do we know wether we hit that limit or not?

At the same time I am reading about massive outages. 30 hours in a month with no announcements made. So you are expecting people to upgrade to an expensive plan to get APIs that aren't slugging away on a stability level that reminds me of products in an alpha stage of development?

With my limited dev time so far I noticed that TS support is spotty at best. This is also mentioned in discussions. To me, I won't even really touch a project that doesn't have decent TS coverage nowadays. If only GraphQL is viable, why is there another client offering anyways? Where does proper TS fall on your roadmap?

I think this leaves much to be desired, especially knowing that the only rational option is to go for the mid-tier level subscription if you want to have a chance of getting half way decent performance and even then, it is not clear to me wether the performance issues are due to geographical location or API throttling or API issues and downtime.

Can you give some transparent information please?

Really not trying to be confrontative here, just want you to be more transparent so we can make informed and rational decisions.

**Comment by greph** [comment, marked as answer]
Hi @genox,

Thank you for voicing your concerns.

The Community plan has limits on API usage, but they're designed to not affect fair use applications. The Standard plan provides higher allowances according to a throttling algorithm. While we can't list specific limits, we're working on providing clearer guidance on avoiding any throttling for your apps. 

Regarding the outages you've mentioned, we're aware of the incidents and have been working diligently to address them. We're making some significant infrastructure changes in order to improve response times globally and have put in additional checks to avoid and better message potential outages.

As for TypeScript support, we're actively toward providing full coverage, as we realize providing this to our users is a key piece of modern development and its a high priority on our product roadmap.

**Reply by genox** [reply]
Thanks for your reply.

> The Community plan has limits on API usage, but they're designed to not affect fair use applications. The Standard plan provides higher allowances according to a throttling algorithm. While we can't list specific limits, we're working on providing clearer guidance on avoiding any throttling for your apps.

Why can't you list specific limits? If those limits are an essential part of the product specification and define how much cost to expect, I'd rather know in advance.

**Reply by ericingram** [reply]
@genox The reason is that it is a dynamic algorithm and we want to reserve the ability to make changes to it quickly as our traffic patterns evolve. Our goal has been to serve as much legitimate traffic as possible, without imposing strict limits that developers have to work around.

On the other hand, it is clear that users want to better understand and plan for traffic that could affect performance. We are currently considering an updated pricing model that would help accomplish this by including usage parameters in pricing.

**Reply by excede1** [reply]
@genox you are quite right to question these things. 
We had a nightmare with Swell, we were in the "Enterprise Plan" and had a team of people relying on their api for headless development and wasted 4 months and a lot of money before we just pulled the pin.
In a nutshell:
- The platform is extremely unreliable
- Many of the so-called features are full of bugs or don't actually work (braintree payments for example)
- Long outages with no notice or an explanation
- Performance is good sometimes and inexplicably terrible at others
- The worst part is that the team and management deny the issues and refuse to acknowledge any fault

Lastly, while it may look good and pretty well resolved from a design perspective (I acknowledge that), it's a toy!
Built and architected on deprecated puigins, single threaded, badly written, untested code (javascript + nodejs).

In the end we are building our own system in golang, fully compiled, tested, multi-threaded, memory efficient, infinitely faster and more reliable than swell. There is just no comparison.
